src/entities/addLiquidityBoosted/index.ts

```diff
@@ -56,27 +58,43 @@ export class AddLiquidityBoostedV3 {
         let bptOut: TokenAmount;
         let amountsIn: TokenAmount[];
 
-        // Child tokens are the lowest most tokens. This will be underlying if it exists.
-        const childTokens = poolState.tokens.map((t) => {
-            if (t.underlyingToken) {
-                return t.underlyingToken;
-            }
-            return {
-                address: t.address,
-                decimals: t.decimals,
-                index: t.index,
-            };
-        });
-
         switch (input.kind) {
             case AddLiquidityKind.Unbalanced: {
+                // Use poolState to add token index to amountsIn
+                const tokensIn = input.amountsIn.map((amountIn) => {
```

Comment:
It feels like a lot of this `tokensIn` logic is repeated within unbalanced and proportional implementations. Do you think it's worth extracting/reusing?

Reply:
given we're in a bit of a tight schedule, we can leave this refactor to be done in a new (low-priority) PR

Reply:
yup I agree on both notes :+1: